#!/usr/bin/env python
PACKAGE = "vehicle_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("angle_p_gain",    double_t,    0, "P gain 4 angle control in differential drive controller", 1.2,  0.0, 10.0)
gen.add("angle_d_gain",    double_t,    0, "D gain 4 angle control in differential drive controller", 0.5,  -10.0, 10.0)
gen.add("position_p_gain", double_t,    0, "P gain 4 position control in differential drive controller", 5.0,  0.0, 10.0)
gen.add("position_d_gain", double_t,    0, "D gain 4 position control in differential drive controller", 0.0,  -10.0, 10.0)
gen.add("speed_p_gain", double_t,    0, "P gain on position error for vehicle speed if path contains time stamps", 1.5,  0.0, 10.0)
gen.add("speed", double_t,    0, "Desired speed 4 position control in differential drive controller", 0.075,  0.0, 1.0)
gen.add("speed_reduction_gain", double_t, 0, "Gain to control the linear speed reduction for rising angular rate", 1.0, 0.0, 100.0)
gen.add("use_final_twist", bool_t,    0, "Flag determining if orientation is corrected at the end of a path", True)
gen.add("final_twist_trials_max", int_t,  0, "Maximum number of control trials for final position approach", 1000, 0, 2000)
gen.add("flipper_low_position", double_t, 0, "Low position of the flippers for driving", 0.1, 0, 0.75)
gen.add("flipper_high_position", double_t, 0, "High position of the flippers for driving", 0.4, 0, 0.75)
gen.add("flipper_switch_position", double_t, 0, "Flipper toggle command: set to high position if its below this value, else to the low", 0.25, 0, 0.75)

gen.add("use_affw", bool_t, 0, "Changes the topic cmd_vel is published to, so affw tracker can be used", False)
gen.add("use_ekf", bool_t, 0, "Use ekf to correct velocity commands", False)

exit(gen.generate(PACKAGE, "vehicle_controller", "PdParams"))
